core/lib/libtomcrypt/authenc.c

```diff
@@ -0,0 +1,439 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2014-2019, Linaro Limited
+ */
+
+#include <assert.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <tomcrypt.h>
+#include <util.h>
+
+/******************************************************************************
+ * Authenticated encryption
+ ******************************************************************************/
+
+#define TEE_CCM_KEY_MAX_LENGTH		32
+#define TEE_CCM_NONCE_MAX_LENGTH	13
+#define TEE_CCM_TAG_MAX_LENGTH		16
+#define TEE_GCM_TAG_MAX_LENGTH		16
+
+#if defined(CFG_CRYPTO_CCM)
```

Comment:
Here also, I think two files would be cleaner (`ccm.c` and `gcm_from_cryptolib.c` to reflect the config names?). Not a strong feeling either.